server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -391,14 +392,16 @@ public String executor() {
                 new InitializeExtensionRequest(transportService.getLocalNode(), extension),
                 initializeExtensionResponseHandler
             );
-            // TODO: make asynchronous
-            inProgressFuture.get(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+            inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+            inProgressFuture.whenComplete((r, e) -> {
```

Comment:
The problem with this flow (and others like that) is that `inProgressFuture`'s result is never asked for. Essentially, the future may complete (or timeout) at some point in time, long after the `initializeExtension` is returned. The call to `get` or `join` is required to actually ensure the computation is finished:

```
inProgressFuture.join();
```

Also, the `whenComplete` is not needed (since it has nothing to do). The `join()` would thrown `CompletionException` (with the cause being set to real exception, including `TimeoutException`) or `CancellationException` if the computation was cancelled.

Reply:
The ExtensionManager does maintain a list of all extensions and a separate list of initialized ones, so it's probably okay not to do anything other than logging on failure at this point.  That may change in the future when we implement hot swapping, etc.

@ryanbogan and I did have a discussion about whether calling `get()` and handling the exception was the right way to go about this, and based on this comment, I think that is appropriate here.